newsroom/mgmt_api/factory.py

```diff
@@ -0,0 +1,126 @@
+#!/usr/bin/env python
+# -*- coding: utf-8; -*-
+#
+# This file is part of Newsroom.
+#
+# Copyright 2021 Sourcefabric z.u. and contributors.
+#
+# For the full copyright and license information, please see the
+# AUTHORS and LICENSE files distributed with this source code, or
+# at https://www.sourcefabric.org/superdesk/license
+
+import os
+import logging
+
+from elasticsearch.exceptions import RequestError as ElasticRequestError
+from werkzeug.exceptions import HTTPException
+from superdesk.errors import SuperdeskApiError
+from superdesk.flask import Config, jsonify
+
+from newsroom.factory import BaseNewsroomApp
+from newsroom.mgmt_api.auth.jwt import JWTTokenAuth
+
+
+logger = logging.getLogger(__name__)
+
+API_DIR = os.path.abspath(os.path.dirname(__file__))
+
+
+class NewsroomMGMTAPI(BaseNewsroomApp):
+    INSTANCE_CONFIG = "settings_mgmtapi.py"
+    AUTH_SERVICE = JWTTokenAuth()
+
+    def __init__(self, import_name=__package__, config=None, **kwargs):
+        if not hasattr(self, "settings"):
+            self.settings = Config(".")
+
+        if config and config.get("BEHAVE"):
+            # ``superdesk.tests.update_config`` adds ``planning`` to ``INSTALLED_APPS``
+            # So if we're running behave tests, reset this config here
+            config["INSTALLED_APPS"] = []
+
+        self.wsgi = self
```

Comment:
What does this line do?

Reply:
It ensures that self.wsgi references the app instance itself, allowing code (like [config_oauth](https://github.com/superdesk/newsroom-core/blob/async/newsroom/auth_server/oauth2.py#L65-L80)) to access `app.wsgi.config`. so this maintains compatibility where wsgi is expected.